external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/package.scala

```diff
@@ -16,9 +16,58 @@
  */
 package org.apache.spark.sql
 
+import java.util.concurrent.TimeUnit
+
 import org.apache.kafka.common.TopicPartition
 
+import org.apache.spark.internal.config.ConfigBuilder
+
 package object kafka010 {   // scalastyle:ignore
   // ^^ scalastyle:ignore is for ignoring warnings about digits in package name
   type PartitionOffsetMap = Map[TopicPartition, Long]
-}
+
+  private[spark] val PRODUCER_CACHE_TIMEOUT =
+    ConfigBuilder("spark.kafka.producer.cache.timeout")
+      .doc("The time to remove the producer when the producer is not used.")
+      .timeConf(TimeUnit.MILLISECONDS)
+      .createWithDefaultString("600000")
+
+  val CONSUMER_POLL_TIMEOUT =
+    ConfigBuilder("kafkaConsumer.pollTimeoutMs")
+      .doc("The timeout in milliseconds to poll data from Kafka in executors.")
+      .timeConf(TimeUnit.MILLISECONDS)
+      .createWithDefaultString("512")
+
+  val CONSUMER_CACHE_CAPACITY =
+    ConfigBuilder("spark.sql.kafkaConsumerCache.capacity")
+      .doc("The size of LinkedHashMap for caching kafkaConsumers.")
+      .intConf
+      .createWithDefaultString("64")
```

Comment:
Why `createWithDefaultString` is used everywhere? For `intConf` you can specify an `Int` by using `createWithDefault()`, and so on. 

As `createWithDefault` is type-safe it is less error prone.